activerecord/test/activejob/destroy_association_later_test.rb

```diff
@@ -0,0 +1,234 @@
+# frozen_string_literal: true
+
+require "activejob/helper"
+
+require "models/book_destroy_later"
+require "models/essay_destroy_later"
+require "models/tag"
+require "models/tagging"
+require "models/essay"
+require "models/category"
+require "models/post"
+require "models/content"
+require "models/destroy_later_parent"
+require "models/destroy_later_parent_soft_delete"
+require "models/dl_keyed_belongs_to"
+require "models/dl_keyed_belongs_to_soft_delete"
+require "models/dl_keyed_has_one"
+require "models/dl_keyed_join"
+require "models/dl_keyed_has_many"
+require "models/dl_keyed_has_many_through"
+
+class DestroyAssociationLaterTest < ActiveRecord::TestCase
+  include ActiveJob::TestHelper
+
+  test "destroying a record destroys the has_many :through records using a job" do
+    tag = Tag.create!(name: "Der be treasure")
+    tag2 = Tag.create!(name: "Der be rum")
+    book = BookDestroyLater.create!
+    book.tags << [tag, tag2]
+    book.save!
+
+    book.destroy
+
+    assert_difference -> { Tag.count }, -2 do
+      perform_enqueued_jobs only: ActiveRecord::DestroyAssociationLaterJob
+    end
+  end
+
+  test "destroying a scoped has_many through only deletes within the scope deleted" do
+    tag = Tag.create!(name: "Der be treasure")
+    tag2 = Tag.create!(name: "Der be rum")
+    parent = BookDestroyLaterWithScopedTags.create!
+    parent.tags << [tag, tag2]
+    parent.save!
+
+    parent.reload # force the association to be reloaded
+
+    parent.destroy
+
+    assert_difference -> { Tag.count }, -1 do
+      perform_enqueued_jobs only: ActiveRecord::DestroyAssociationLaterJob
+    end
+    assert_raises ActiveRecord::RecordNotFound do
+      tag2.reload
+    end
+    assert tag.reload
+  end
+
+  test "enqueues the has_many through to be deleted with custom primary key" do
```

Comment:
This test ensures that we can find and delete an association with a custom primary key:
```
 self.primary_key = "many_key"
```